### DISCUSSION TITLE
Rearrangeable Columns in Tables

### DISCUSSION THREAD

**Discussion by Fazle-Jahangir** [twilio-labs/paste#2332] [General] [answered]
**Background**

Currently in the exploration phase of adding a feature to Insights Queue Stats where users are able to show or hide columns in a table. This brings about other questions such as changing the order of the columns so that the most important metrics in insights are first. 

**Question** 

Can you share your thoughts on the behaviour of data grids? Is it possible to rearrange the columns and if not how would you feel about exploring this option in Flex Insights? At the moment the queue stats display has no controls within it. 

<img width="1020" alt="Screenshot 2022-03-30 at 12 15 33" src="https://user-images.githubusercontent.com/100706574/160824132-ed280f2f-5d19-428a-8b7d-2166ce614002.png">

**Comment by shleewhite** [comment, marked as answer]
Hi @Fazle-Jahangir!

Unfortunately, rearrangeable columns is out of scope for the DataGrid component and isn't on our roadmap. 

One workaround, is add a table actions button that launches a Popover that has inputs where users can set the number the column should be in the table and/or checkboxes that users can toggle to display the column. That way, when you save the Popover, the DataGrid can re-render to match the user's preferences.

Could this solution work in your use case?

**Reply by Fazle-Jahangir** [reply]
Hi Lee, 

Thank you for sharing this workaround. I have had a look at data grids on Paste and found this example of a type of popover you may be referring to. Do you know at all what clicking on setting does in this use case? Does it launch a sub-menu within the popover or open a modal? This would help me further understand on how a popover could work in my use case. 

<img width="315" alt="Screenshot 2022-04-04 at 12 47 09" src="https://user-images.githubusercontent.com/100706574/161548007-9a8fbaa9-2f5e-4bc1-97bd-7ac64bf2f84d.png">

**Reply by andioneto** [reply]
@Fazle-Jahangir Hello there, I think we can move this discussion over to [Discussion 2338](https://github.com/twilio-labs/paste/discussions/2338), since that is in fact what we are suggesting. Head over to the above discussion for more info. Thanks!
